### PR TITLE
Add a class that wraps all Elasticsearch operations + wrap job execution into a class

### PR DESCRIPTION
**Pull request by artem-shelkovnikov**
Several changes bundled in 1 PR 🙈 

1. Add `Framework::ElasticConnectorActions` class that wraps all domain connector calls to Elastisearch. This class can and should be used to define our interface with Kibana in the end - all methods with description/explanation at one place.
2. Add `Framework::ConnectorSettings` convenience wrapper that wraps connector settings in an abstraction. I think it makes work with config more abstract, and at the same more structured - instead of operating with object that contains response from Elasticsearch, we will operate on a structured object.
3. Add Framework::SyncJobRunner to encapsulate the logic of actually running the job and managing the job status.

### CODE REVIEW

File: lib/framework/connector_settings.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      @elasticsearch_response[:_source][:configuration]
+    end
+
+    def scheduling_settings
+      self['scheduling']
+    end
+
+    def configuration_initialized?
+      return false if configuration.nil?
+      configuration.present?
```
[the remark this answers is not in the document]
Comment (artem-shelkovnikov) [reply lib/framework/connector_settings.rb:35]: It's actually coming from https://api.rubyonrails.org/v4.2/classes/Object.html, namespace is:
`active_support/core_ext/object/blank`, adding it as a require statement, thanks!

We also have usage of `present?` all over the project - might wanna reconsider having whole ActiveSupport gem for such a small slice of functionality?

File: lib/app/connector.rb [hunk 2]
```diff
@@ -91,89 +57,16 @@ def start_polling_jobs
...
-
-        @client.update(:index => connector['_index'], :id => connector['_id'], :body => body)
-        if error
-          Utility::Logger.info("Failed to sync for connector #{connector['_id']} with error #{error}")
-        else
-          Utility::Logger.info("Successfully synced for connector #{connector['_id']}")
-        end
+        Framework::SyncJobRunner.new(
+          connector_settings: connector_settings,
+          connector_instance: connector_instance
```
[the remark this answers is not in the document]
Comment (artem-shelkovnikov) [reply lib/app/connector.rb:67]: Good call, I was doing a reguilar inversion-of-control coding here, but reflecting on it it feels like it's not working very well in languages with no strong typing, just making things more confusing.

I will move the initialization into the SyncJobRunner

**Review by mchernyavskaya — APPROVED** [review]

**Comment by vidok** [comment]
@artem-shelkovnikov I see you introduced a new namespace `Framework` so I'm lost now :). Could you please give more context to help me understand the difference between `framework` and `utility` namespaces?

**Comment by wangch079** [comment]
>@artem-shelkovnikov I see you introduced a new namespace Framework so I'm lost now :). Could you please give more context to help me understand the difference between framework and utility namespaces?

@vidok , `utility` namespace means.... `utilities`, e.g. logger, monitor, exception handler, etc...

I think what @artem-shelkovnikov was trying to add is some framework-level stuff, thus, the namespace `framework`, which definitely should not fall into `utility`. I also felt a bit strange about having `framework` but I gave up because I don't have any better suggestions.

**Comment by vidok** [comment]
> > @artem-shelkovnikov I see you introduced a new namespace Framework so I'm lost now :). Could you please give more context to help me understand the difference between framework and utility namespaces?
> 
> @vidok , `utility` namespace means.... `utilities`, e.g. logger, monitor, exception handler, etc...
> 
> I think what @artem-shelkovnikov was trying to add is some framework-level stuff, thus, the namespace `framework`, which definitely should not fall into `utility`. I also felt a bit strange about having `framework` but I gave up because I don't have any better suggestions.

Thanks for you reply. 

I'm asking myself the following questions: 
* What components are being used in connectors service and connectors? 
* Do we expect our community to use sharable components when they write a new connector in Ruby?

If both answers are `Yes` then I'd love to have a namespace (even a gem) which is called `connectors_sdk`. All shared components can leave there: logger, monitor, exceptions handler, connectors abstractions. I think it's the best name for it because it's a software development kit :)

**Comment by wangch079** [comment]
> I'd love to have a namespace (even a gem) which is called `connectors_sdk`. All shared components can leave there: logger, monitor, exceptions handler, connectors abstractions.

Guess what, our `connectors` namespace used to be `connector_sdk`, and `utility` used to be `connectors_shared`. @artem-shelkovnikov did the renaming in https://github.com/elastic/connectors/pull/127

**Comment by vidok** [comment]
> > I'd love to have a namespace (even a gem) which is called `connectors_sdk`. All shared components can leave there: logger, monitor, exceptions handler, connectors abstractions.
> 
> Guess what, our `connectors` namespace used to be `connector_sdk`, and `utility` used to be `connectors_shared`. @artem-shelkovnikov did the renaming in #127

Ahh, I'd love to have a look at the Slack thread. Do you have a link?

**Comment by wangch079** [comment]
@vidok [this](https://elastic.slack.com/archives/C03GAPTQ2JF/p1655906456162229)?

**Comment by danajuratoni** [comment]
> should we allow creating new connectors via CLI
It's definitely a nice feature, especially if it doesn't require a huge amount of effort to develop and maintain. 
Could we do a quick research to determine how it could evolve, the impact of adding new connectors functionality (e.g. indexing rules, sync job management, etc.) and how would tracking look like? By tracking I mean identifying who performed an action (Kibana or CLI).

**Comment by wangch079** [comment]
>Could we do a quick research to determine how it could evolve, the impact of adding new connectors functionality (e.g. indexing rules, sync job management, etc.)

I don't think parity between Kibana and CLI is necessary. e.g. I don't see a need to specify indexing rules in CLI. For job scheduling, a force sync is good enough in CLI, but not scheduling management.

>how would tracking look like? By tracking I mean identifying who performed an action (Kibana or CLI).

We currently won't be able to track this.

**Comment by wangch079** [comment]
> If this will be customer facing though, we need clear documentation for any of the following options:
> 
> 1. CLI offered as part of the Connector Service tech preview -> document what is supported, what not and what is the consequence of mixing CLI and Kibana. In this case, tracking would not be optional. If this is not feasible for 8.4, I would suggest to consider option 2.
> 2. CLI marked as experimental / "use at your own peril" -> document what it was intended for and possible issues if using it.

@artem-shelkovnikov @mchernyavskaya WDYT? I prefer option 1 because I think we need tracking eventually and it's not a big effort.

**Comment by artem-shelkovnikov** [comment]
@wangch079 @danajuratoni I'm also leaning towards option 1.

**Pull request merged**